From 26638c4f91381a4c3a1b069e152c93d6e961bdd7 Mon Sep 17 00:00:00 2001
From: Youngsoo Choi <youngsoo.choi@lge.com>
Date: Mon, 7 Sep 2020 17:40:10 +0900
Subject: [PATCH] [op][ds][build] Fix build error when target is arm64

:Release Notes:
Fix build error when target is arm64

:Detailed Notes:
When build target is arm64(e.g. SA8155, RPi4-64) there is an error on
libjpeg_turbo library as below.

../../../../../../../../../../../../home2/drabble/webOS/Liberty/Upgradeto84/gpro/auto-work/chromium84-ose/src/third_party/libjpeg_turbo/simd/arm/common/jidctint-neon.c:217:30:
error: incompatible types when initializing type 'int64_t' {aka 'long int'} using type 'int64x1_t'
|   217 |   int64_t bitmap_rows_4567 = vreinterpret_s64_s16(bitmap);

It was impossible to remove only the files which occurs error with NEON
instructions remove all files, which means drop NEON instruction support
from libjpeg_turbo on arm64, as a work around.

:Testing Performed:
Build succeeds

:QA Notes:
Test-cases: N/A

:Issues Addressed:
[PLAT-115645] Prepare OSE/Auto build

Change-Id: I2cc50d0b22bca9e21b3aa7e8e2b2e8e22bf97933
Reviewed-on: http://gpro.lge.com/c/webosauto/chromium84/+/285079
Reviewed-by: Commit Msg Checker <commit_msg@lge.com>
Reviewed-by: Ban Word Checker <ban_word@lge.com>
Reviewed-by: Youngsoo Choi <youngsoo.choi@lge.com>
Tested-by: Youngsoo Choi <youngsoo.choi@lge.com>
---
 src/third_party/libjpeg_turbo/BUILD.gn | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/src/third_party/libjpeg_turbo/BUILD.gn b/src/third_party/libjpeg_turbo/BUILD.gn
index 20e95f3a46..0c112fab33 100644
--- a/src/third_party/libjpeg_turbo/BUILD.gn
+++ b/src/third_party/libjpeg_turbo/BUILD.gn
@@ -164,19 +164,6 @@ static_library("simd") {
     ]
     configs -= [ "//build/config/compiler:default_optimization" ]
     configs += [ "//build/config/compiler:optimize_speed" ]
-  } else if (current_cpu == "arm64") {
-    sources = [
-      "simd/arm/arm64/jsimd.c",
-      "simd/arm/arm64/jsimd_neon.S",
-      "simd/arm/common/jdcolor-neon.c",
-      "simd/arm/common/jdmerge-neon.c",
-      "simd/arm/common/jdsample-neon.c",
-      "simd/arm/common/jidctfst-neon.c",
-      "simd/arm/common/jidctint-neon.c",
-      "simd/arm/common/jidctred-neon.c",
-    ]
-    configs -= [ "//build/config/compiler:default_optimization" ]
-    configs += [ "//build/config/compiler:optimize_speed" ]
   } else {
     sources = [
       "jsimd_none.c",
-- 
2.17.1

